diesel_cli/src/main.rs

```diff
@@ -210,9 +226,15 @@ fn create_migrations_dir(matches: &ArgMatches) -> DatabaseResult<PathBuf> {
         .map(PathBuf::from)
         .or_else(|| env::var("MIGRATION_DIRECTORY").map(PathBuf::from).ok())
         .unwrap_or_else(|| {
-            find_project_root()
-                .unwrap_or_else(handle_error)
-                .join("migrations")
+            let migrations_dir_from_config = migrations_dir_from_config(matches);
```

Comment:
Possibly lift out the common component between `migrations_dir` and this function instead of adding to this function? But not necessary.

Reply:
What do you mean by the common component?

Reply:
```rust
fn migrations_dir(matches: &ArgMatches) -> PathBuf {
    migrations_dir_from_cli(matches)
        .or_else(|| env::var("MIGRATION_DIRECTORY").map(PathBuf::from).ok())
        .or_else(|| migrations_dir_from_config(matches).ok())
        .unwrap_or_else(|| migrations::find_migrations_directory().unwrap_or_else(handle_error))
}

// -- snip

fn create_migrations_dir(matches: &ArgMatches) -> DatabaseResult<PathBuf> {
    // This statment seems to be (almost) a c+p of the `migrations_dir` method.
    // That is it looks like it could be `let dir = matches.migrations_dir()` instead.
   // If this isn't the case, then can we refactor a little so that the `migration_dir` can
   // be directly called instead?
    let dir = matches
        .value_of("MIGRATION_DIRECTORY")
        .map(PathBuf::from)
        .or_else(|| env::var("MIGRATION_DIRECTORY").map(PathBuf::from).ok())
        .unwrap_or_else(|| {
            let migrations_dir_from_config = migrations_dir_from_config(matches);
            let project_root = find_project_root().unwrap_or_else(handle_error);

            // Retreives and transforms the path of the migrations
            // directory from the configuration file or returns the
            // default one.
            migrations_dir_from_config
                .ok()
                .unwrap_or_else(|| project_root.join("migrations"))
        });

    // -- snip
}
```

Reply:
I wrote it in a comment in the code block, but:

The statement in `create_migrations_dir` looks really similar to the `migrations_dir` function, so can we call `migrations_dir` instead of typing it out again?

If this isn't the case, then can we refactor a little so that this can happen?

Reply:
Ok thank you for the details. I will see what I can do soon.